api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -132,7 +136,8 @@ public Cohort(String commaSeparatedIds) {
 		this();
 		for (StringTokenizer st = new StringTokenizer(commaSeparatedIds, ","); st.hasMoreTokens();) {
 			String id = st.nextToken();
-			memberIds.add(Integer.valueOf(id.trim()));
+			Patient pid = new Patient(Integer.valueOf(id.trim()));
+            members.add(new CohortMembership(pid));
```

Comment:
You should call `addMember(CohortMembership)`, you should always call addMember() instead of adding the  members directly to the collection, adding the member to the list should ONLY happen in addMember(CohortMembership) method which should be responsible for setting the cohort field on the membership